### PR TITLE
[react] Local Wallet UI refactor, various fixes and cleanup

### PR DESCRIPTION
**Pull request by MananTank**
* Remove Guest Mode flag
* Remove credentialStorage flow
* Allow two modes - persist - `true` and `false`
* Fix Safe + Magic Link not working
* use walletIds object instead of using raw strings for wallet id
* Fix EOA wallets using wrong config in smartwallet or safe
* Fix opening the connect UI of wallet directly if there's a single wallet

### CODE REVIEW

File: packages/react-core/src/core/providers/thirdweb-provider.tsx [hunk 1]
```diff
@@ -53,13 +53,6 @@ export interface ThirdwebProviderCoreProps<TChains extends Chain[]>
    */
   autoConnect?: boolean;
 
-  /**
-   * Whether or not to show a "Join as Guest" option in the wallet modal.
-   *
-   * Join as guest will connect the user to the dApp using a guess wallet users can export.
-   */
-  guestMode?: boolean;
```
Comment (iketw) [inline packages/react-core/src/core/providers/thirdweb-provider.tsx:61]: Thanks for removing!

**Review by iketw — APPROVED** [review]
Do we need to add a changeset?

File: packages/wallets/src/evm/walletIds.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const walletIds = {
+  coinbase: "coinbaseWallet",
+  metamask: "metamask",
+  localWallet: "localWallet",
+  smartWallet: "SmartWallet",
+  safe: "Safe",
```
Comment (iketw) [inline packages/wallets/src/evm/walletIds.ts:6]: should have a consistent naming pattern here? All others start with lower case, we can rename these two to `safe` and `smartWallet`

File: packages/react-core/src/core/types/wallet.ts [hunk 3]
```diff
@@ -23,11 +23,12 @@ export type WalletClass<I extends WalletInstance = WalletInstance> = {
   meta: (typeof AbstractClientWallet)["meta"];
 };
 
-export type Wallet<I extends WalletInstance = WalletInstance> = {
+export type Wallet<
+  I extends WalletInstance = WalletInstance,
+  Config extends Record<string, any> = Record<string, any>,
```
[the remark this answers is not in the document]
Comment (MananTank) [reply packages/react-core/src/core/types/wallet.ts]: done

File: packages/wallets/src/evm/walletIds.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export const walletIds = {
```
Comment (iketw) [inline packages/wallets/src/evm/walletIds.ts:1]: There's a `constants` folder in this path, it's probably worth it to move this file in there

Comment (MananTank) [reply packages/wallets/src/evm/walletIds.ts]: done

**Pull request merged**